public/chat_header_button.tsx

```diff
@@ -179,6 +182,7 @@ export const HeaderChatButton = (props: HeaderChatButtonProps) => {
               )}
             </span>
           }
+          disabled={!props.userHasAccess}
```

Comment:
Have you tested if the click handler on line 161 is also disabled?

Reply:
We only disable the chat input in this PR. Do we need to disable the dashboard assistant icon? I think the icon still can be clicked before the chat input added.

Reply:
Then user can still click the icon to display the flyout, that also makes sense to me.